src/components/app/App.test.tsx

```diff
@@ -0,0 +1,290 @@
+            });
+
+            act(() => {
+                render(
+                    <MemoryRouter initialEntries={['']}>
+                        <App />
+                    </MemoryRouter>
+                )
+            });
+
+            const loadingArticles = await screen.findByText(/searching for articles/i);
+            const loadingSections = await screen.findByText(/finding sections/i);
+
+            waitFor(() => expect(loadingArticles).toBeInTheDocument());
+            waitFor(() => expect(loadingSections).toBeInTheDocument());
+        });
+    });
+
+    describe(`requests successful`, () => {
+        it(`should render complete App component`, async () => {
+            jest.spyOn(httpGetModule, "httpGet").mockImplementation((url: string, params?: Record<string, string>) => {
+                return (url === 'sections') ? Promise.resolve(mockedSectionsResponse) : Promise.resolve(mockedSearchResponse);
+            });
+
+            await act(async () => {
+                render(
+                    <MemoryRouter initialEntries={['']}>
+                        <App />
+                    </MemoryRouter>
+                );
+            });
+
+            const searchForm = screen.getByRole('form');
+            const searchbox = screen.getByRole('searchbox', { name: /search/i });
+            const submit = screen.getByRole('button', { name: /find/i });
+            const resetArticles = screen.getByRole('button', { name: /all articles/i });
+            const resetSections = screen.getByRole('button', { name: /all sections/i });
+            const sectionsBtn = screen.getByRole('button', { name: /^sections$/i })
+            const sectionsList = screen.getByRole('list', { name: /sections list/i });
+            const sectionsBtns = screen.getAllByText(/test-section-webTitle/i);
+            const articles = screen.getAllByRole('link');
+            const elementPrev = screen.getByText(/previous/i);
+            const elementNext = screen.getByText(/next/i);
+            const expectedStartValue = screen.getByText('1')
+            const expectedEndValue = screen.getByText('10')
+
+            expect(searchForm).toBeInTheDocument();
+            expect(searchbox).toBeInTheDocument();
+            expect(submit).toBeInTheDocument();
+            expect(resetArticles).toBeInTheDocument();
+            expect(resetSections).toBeInTheDocument();
+            expect(sectionsBtn).toBeInTheDocument();
+            expect(sectionsList).toBeInTheDocument();
+            expect(sectionsBtns).toHaveLength(20);
+            expect(articles).toHaveLength(10);
+            expect(elementPrev).toBeInTheDocument();
+            expect(elementNext).toBeInTheDocument();
+            expect(expectedStartValue).toBeInTheDocument();
+            expect(expectedEndValue).toBeInTheDocument();
+        });
+    });
+
+    describe(`on Change event on searchbox`, () => {
```

Comment:
It all looks like huge integration tests instead unit tests. All of this is happening in inner components so should not be tested here. Responsibility of this component is mostly about getting the data and reacting to events in inner components.
So for example here I would not test if something is displayed in inner component but if something is dane if change event occurs -> like fetching new data.
And basically this comment is also for other tests in this file.